### PR TITLE
fix(compiler-cli): use '' for the source map URL of indirect templates

### LINKED ISSUE #40854: Angular 11.1.1 breaks source map for components with variable as an inline template
# 🐞 bug report

### Affected Package
@angular/compiler-cli

### Is this a regression?
 Yes. 11.1.0 does not present the issue.

### Description

This component source file
```
// test.component.ts

import { Component, OnInit } from '@angular/core';

const x = '<p>test works!</p>';
@Component({
  selector: 'app-test',
  template: x
})
export class TestComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }
}
```
when compiled with 11.1.1 looks like this in chrome

bad

instead of this

good

## 🔬 Minimal Reproduction
https://github.com/angelyordanov/angular-template-problem

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 11.1.1
Node: 14.4.0
OS: darwin x64
</code></pre>

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by alxhub**
Indirect templates are templates produced by a non-literal expression value
of the `template` field in `@Component`. The compiler can statically
determine the template string, but there is not guaranteed to be a physical
file which contains the bytes of the template string. For example, the
template string may be computed by a concatenation expression: 'a' + 'b'.

Previously, the compiler would use the TS file path as the source map path
for indirect templates. This is incorrect, however, and breaks source
mapping for such templates, since the offsets within the template string do
not correspond to bytes of the TS file.

This commit returns the compiler to its old behavior for indirect templates,
which is to use `''` as the source map URL for such templates.

Fixes #40854

### CODE REVIEW

File: packages/compiler-cli/src/ngtsc/annotations/src/component.ts [hunk 1]
```diff
@@ -1133,11 +1141,11 @@ export class ComponentDecoratorHandler implements
 
   private _parseTemplate(
       template: TemplateDeclaration, sourceStr: string, sourceParseRange: LexerRange|null,
-      escapedString: boolean): ParsedComponentTemplate {
+      escapedString: boolean, sourceMapUrl: string|null): ParsedComponentTemplate {
     // We always normalize line endings if the template has been escaped (i.e. is inline).
     const i18nNormalizeLineEndingsInICUs = escapedString || this.i18nNormalizeLineEndingsInICUs;
 
-    const parsedTemplate = parseTemplate(sourceStr, template.sourceMapUrl, {
+    const parsedTemplate = parseTemplate(sourceStr, template.potentialSourceMapUrl, {
```
Comment (petebacondarwin) [inline packages/compiler-cli/src/ngtsc/annotations/src/component.ts:1148]: Should this be:

```suggestion
    const parsedTemplate = parseTemplate(sourceStr, sourceMapUrl ?? '', {
```

similar to the other calls to `parseTemplate()`?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
I think we could drop the `potentialSourceMapUrl` altogether, since the information needed is already encoded in the `ParsedTemplateWithSource` interface via the `resolvedTemplateUrl`.

(Note that there is no longer any need to call `sourceMapUrl(resourceUrl)` to get the sourceMapUrl, since the TS bug was fixed in TS 3.3 and we should also remove that helper and the check `tsSourceMapBug29300Fixed()`.)

File: packages/compiler-cli/src/ngtsc/annotations/src/component.ts (hunk 1 above)
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/annotations/src/component.ts]: Yes indeed - fixed.

**Comment by alxhub** [comment]
> I think we could drop the `potentialSourceMapUrl` altogether, since the information needed is already encoded in the `ParsedTemplateWithSource` interface via the `resolvedTemplateUrl`.
> 
> (Note that there is no longer any need to call `sourceMapUrl(resourceUrl)` to get the sourceMapUrl, since the TS bug was fixed in TS 3.3 and we should also remove that helper and the check `tsSourceMapBug29300Fixed()`.)

Agreed - we can do this in a future cleanup, to avoid mixing refactorings in with the bug fix.

**Review by petebacondarwin — APPROVED** [review]
I can knock up the refactoring once this has landed.

**Pull request closed without merging**